### PR TITLE
Added swagger-jsdoc

### PR DESCRIPTION
**Pull request by joshmorel**
Tags is currently just a todo but I can add later.

I put the param/path docs in controllers because that is where the existing documentation is and I figured we didn't want to do two kinds of documentation. The definition docs are in the models.

The one issue with this approach is that the actual path name is in the route files. I was wondering, not just for this reason but more so for simpler organization, does it make sense to co-locate the routes and controllers?

### CODE REVIEW

**Comment by jasonify** [comment]
Wow! This looks amazing. Just pulled it locally 💯  🥇

**Comment by jasonify** [comment]
Re: co-locate routes + controllers, can you give an very simple example of what that would look like? I am not exactly sure what you mean.

**Comment by TheHollidayInn** [comment]
We've been talking in the Architecture issue about a different organization. @joshmorel said he will be posting a PR example.

File: server/controllers/comment.controller.js [hunk 1]
```diff
@@ -22,22 +51,41 @@ function create(req, res, next) {
   comment.author = user._id
   comment.save()
   .then((commentSaved)  => {
+    // TODO: result key is not consistent with other responses, consider changing this
```
Comment (jasonify) [inline server/controllers/comment.controller.js]: Ah yeah.  My thinking was doing it this way could allow us to add meta data later on.  For example 'cursors' when doing list type requests.  Sadly it's inconsistent and provides no benefit right now.

Comment (joshmorel) [reply server/controllers/comment.controller.js]: It seemed like a good idea but I was wondering about what the best practice was. Digital Ocean is often lauded for the amazing API and they seem to do [something similar](https://developers.digitalocean.com/documentation/v2/#list-all-droplets). If you think it's a good idea, we should keep it and a long-term todo refactor could be migrating everything else to be consistent with this idea.

Comment (TheHollidayInn) [reply server/controllers/comment.controller.js]: Why don't we just use the standard data key return? We can always add other meta data as well.

Comment (jasonify) [reply server/controllers/comment.controller.js]: I am not sure what you mean?  Either way, it's probably not a big deal to have the comments endpoint  match the other ones.

**Comment by TheHollidayInn** [comment]
Alright, looks great!

**Comment by jasonify** [comment]
This looks great to me. Merging.

**Pull request merged**